BrowserKit/Sources/MenuKit/MenuRedesign/MenuRedesignMainView.swift

```diff
@@ -10,42 +10,72 @@ public final class MenuRedesignMainView: UIView,
                                          ThemeApplicable {
     private struct UX {
         static let headerTopMargin: CGFloat = 15
+        static let horizontalMargin: CGFloat = 16
+        static let closeButtonSize: CGFloat = 30
     }
 
+    public var closeButtonCallback: (() -> Void)?
+
     // MARK: - UI Elements
     private var tableView: MenuRedesignTableView = .build()
-
-    // MARK: - Initializers
-    override init(frame: CGRect) {
-        super.init(frame: frame)
-        setupView()
+    private lazy var closeButton: CloseButton = .build { button in
+        button.addTarget(self, action: #selector(self.closeTapped), for: .touchUpInside)
     }
 
-    required init?(coder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
-    }
+    private var viewConstraints: [NSLayoutConstraint] = []
 
     // MARK: - UI Setup
-    private func setupView() {
+    private func setupView(with data: [MenuSection]) {
+        self.removeConstraints(viewConstraints)
+        viewConstraints.removeAll()
         self.addSubview(tableView)
-        NSLayoutConstraint.activate([
-            tableView.topAnchor.constraint(equalTo: self.topAnchor, constant: UX.headerTopMargin),
-            tableView.bottomAnchor.constraint(equalTo: self.bottomAnchor),
-            tableView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
-            tableView.trailingAnchor.constraint(equalTo: self.trailingAnchor)
-        ])
+        if let section = data.first(where: { $0.isHomepage }), section.isHomepage {
+            self.addSubview(closeButton)
+            viewConstraints.append(contentsOf: [
+                closeButton.topAnchor.constraint(equalTo: self.topAnchor, constant: UX.headerTopMargin),
+                closeButton.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: -UX.horizontalMargin),
+                closeButton.widthAnchor.constraint(equalToConstant: UX.closeButtonSize),
+                closeButton.heightAnchor.constraint(equalToConstant: UX.closeButtonSize),
+
+                tableView.topAnchor.constraint(equalTo: closeButton.bottomAnchor, constant: UX.headerTopMargin / 2),
```

Comment:
nit: Could be a separate UX constant like` headerTopMarginWithButton = 8` instead of `UX.headerTopMargin / 2`